hls-plugin-api/src/Ide/Types.hs

```diff
@@ -431,304 +414,332 @@ instance PluginRequestMethod TextDocumentCodeAction where
   combineResponses _ _ _ _ (x :| _) = x
 
-instance PluginRequestMethod TextDocumentDocumentHighlight where
+instance PluginRequestMethod Method_TextDocumentDocumentHighlight where
 
-instance PluginRequestMethod TextDocumentReferences where
+instance PluginRequestMethod Method_TextDocumentReferences where
 
-instance PluginRequestMethod WorkspaceSymbol where
+instance PluginRequestMethod Method_WorkspaceSymbol where
+    -- TODO: combine WorkspaceSymbol. Currently all WorkspaceSymbols are dumped
+    -- as it is new of lsp-types 2.0.0.0
+    combineResponses _ _ _ _ xs = InL $ mconcat $ takeLefts $ toList xs
 
-instance PluginRequestMethod TextDocumentCodeLens where
+instance PluginRequestMethod Method_TextDocumentCodeLens where
 
-instance PluginRequestMethod TextDocumentRename where
+instance PluginRequestMethod Method_TextDocumentRename where
 
-instance PluginRequestMethod TextDocumentHover where
-  combineResponses _ _ _ _ (catMaybes . toList -> hs) = h
+instance PluginRequestMethod Method_TextDocumentHover where
+  combineResponses _ _ _ _ (mapMaybe nullToMaybe . toList -> hs :: [Hover]) =
+    if mcontent ^. L.value == ""
+        then InR Null
+        else InL $ Hover (InL mcontent) r
     where
-      r = listToMaybe $ mapMaybe (^. range) hs
-      h = case foldMap (^. contents) hs of
-            HoverContentsMS (List []) -> Nothing
-            hh                        -> Just $ Hover hh r
-
-instance PluginRequestMethod TextDocumentDocumentSymbol where
-  combineResponses _ _ (ClientCapabilities _ tdc _ _ _) params xs = res
+      r = listToMaybe $ mapMaybe (^. L.range) hs
+      -- We are only taking MarkupContent here, because MarkedStrings have been
+      -- deprecated for a while and don't occur in the hls codebase
+      mcontent :: MarkupContent
+      mcontent = mconcat $ takeLefts $ map (^. L.contents) hs
+
+instance PluginRequestMethod Method_TextDocumentDocumentSymbol where
+  combineResponses _ _ (ClientCapabilities _ tdc _ _ _ _) params xs = res
     where
-      uri' = params ^. textDocument . uri
+      uri' = params ^. L.textDocument . L.uri
       supportsHierarchy = Just True == (tdc >>= _documentSymbol >>= _hierarchicalDocumentSymbolSupport)
-      dsOrSi = fmap toEither xs
+      dsOrSi :: [Either [SymbolInformation] [DocumentSymbol]]
+      dsOrSi =  toEither <$> mapMaybe nullToMaybe' (toList xs)
+      res :: [SymbolInformation] |? ([DocumentSymbol] |? Null)
       res
-        | supportsHierarchy = InL $ sconcat $ fmap (either id (fmap siToDs)) dsOrSi
-        | otherwise = InR $ sconcat $ fmap (either (List . concatMap dsToSi) id) dsOrSi
-      siToDs (SymbolInformation name kind _tags dep (Location _uri range) cont)
+        | supportsHierarchy = InR $ InL $ concatMap (either (fmap siToDs) id) dsOrSi
+        | otherwise = InL $ concatMap (either id ( concatMap dsToSi)) dsOrSi
+      -- Is this actually a good conversion? It's what there was before, but some
```

Comment:
no, these are all kind of dodgy, and I'd like to move functions for a lot of these conversions to lsp and test them